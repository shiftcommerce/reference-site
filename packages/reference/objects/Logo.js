// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Objects
import Image from '../objects/Image'

class Logo extends Component {
  render () {
    let { className } = this.props
    let logoSrc = '/static/shopgo-logo.svg'

    return (
      <div className={className}>
        <Link href='/slug?slug=/homepage' as='/' >
          <a>
            <Image width={120} height={30} src={logoSrc} className='o-logo__image' />
          </a>
        </Link>
      </div>
    )
  }
}

export default Logo
