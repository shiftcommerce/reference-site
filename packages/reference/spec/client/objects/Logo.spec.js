// Object
import Logo from '../../../client/objects/Logo'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Logo className='o-header__logo' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
