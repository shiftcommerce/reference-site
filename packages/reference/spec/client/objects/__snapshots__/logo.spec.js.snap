// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Logo
  className="o-header__logo"
>
  <div
    className="o-header__logo"
  >
    <Link
      as="/"
      href="/slug?slug=/homepage"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <Image
          className="o-logo__image"
          height={40}
          src="/static/shopgo-logo.svg"
          width={180}
        >
          <div
            className="o-logo__image"
          >
            <picture>
              <img
                className="o-image"
                height={40}
                src="/static/shopgo-logo.svg"
                srcSet="/static/shopgo-logo.svg 769w"
                width={180}
              />
            </picture>
          </div>
        </Image>
      </a>
    </Link>
  </div>
</Logo>
`;
